src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/CertificateExtensionsCommon.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Internal.Cryptography.Pal
+{
+    internal static class CertificateExtensionsCommon
+    {
+        public static A GetPublicKey<A>(this X509Certificate2 certificate) where A : AsymmetricAlgorithm
```

Comment:
I suggest if we want to use something besides ```T```, we should do so because we want a descrive name, e.g. ```TAlgorithm```.